services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java

```diff
@@ -154,11 +154,18 @@ public String deleteVpcRouter(String projectId, String vpcId) throws Exception {
     }
 
     @Override
-    public RouteTable getOrCreateVpcRouteTable(String projectId, String vpcId) throws DatabasePersistenceException, CanNotFindVpc, CacheException, OwnMultipleVpcRouterException {
+    public RouteTable getVpcRouteTable(String projectId, String vpcId) throws DatabasePersistenceException, CanNotFindVpc, CacheException, OwnMultipleVpcRouterException, CanNotFindRouter {
         RouteTable routeTable = null;
-
-        // Get or create a router for a Vpc
-        Router router = getOrCreateVpcRouter(projectId, vpcId);
+        VpcWebJson vpcResponse = this.vpcRouterToVpcService.getVpcWebJson(projectId, vpcId);
+        VpcEntity vpcEntity = vpcResponse.getNetwork();
+        if (vpcEntity == null)
+        {
+            throw new CanNotFindVpc();
+        }
+        Router router = vpcEntity.getRouter();
+        if (router == null) {
+            throw new CanNotFindRouter();
+        }
```

Comment:
@DavidLiu506 Don't throw exceptions if you cannot find vpc or router in the get function, just `logger.warn` and return null.

Reply:
Talked to James. throw vpc or router not found.

Reply:
agree